inc/options/class-wpseo-option.php

```diff
@@ -320,36 +320,44 @@ public function validate_verification_string( $key, $dirty, $old, &$clean ) {
 
 	/**
 	 * @param string $key   Key to check, by type of service.
-	 * @param array  $dirty Dirty data.
+	 * @param array  $dirty Dirty data with the new values.
 	 * @param array  $old   Old data.
-	 * @param array  $clean Clean data by reference.
+	 * @param array  $clean Clean data by reference, normally the default values.
 	 */
 	public function validate_url( $key, $dirty, $old, &$clean ) {
 		if ( isset( $dirty[ $key ] ) && $dirty[ $key ] !== '' ) {
-			$url = WPSEO_Utils::sanitize_url( $dirty[ $key ] );
-			if ( $url !== '' ) {
-				$clean[ $key ] = $url;
-			}
-			else {
+
+			$submitted_url = trim( htmlspecialchars( $dirty[ $key ] ) );
```

Comment:
Is this the correct function to use for an URL (because it's not HTML)?

Reply:
At this point of the code we just need to trim and do some basic encoding, so that `FILTER_VALIDATE_URL` can do is job. The purpose here is to check if the entered URL has a valid format. A full sanitization happens later.